UOP1_Project/Assets/PlayerSpawnSystem/PlayerLevelChange.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+
+//The main "Spawn System" management class. Call the
+public class PlayerLevelChange : MonoBehaviour
+{
+    public static PlayerLevelChange main;
+    [SerializeField] List<PlayerSpawnPoint> spawnPoints = new List<PlayerSpawnPoint>(); //Contains all the spawn points in the current scene
+
+    void Awake()
+    {
+        //========== Singleton design stuff ==========
+        if (main == null)
+        {
+            DontDestroyOnLoad(gameObject);
+            main = this;
+        }
+        else if (main != this)
+        {
+            Destroy(gameObject);
+        }
+
+        //========== Initialization ==========
+        SceneManager.sceneLoaded += OnSceneLoaded;
+    }
+
+    void OnSceneLoaded(Scene scene, LoadSceneMode mode)
+    {
+        ClearSpawnPoints();
+    }
+
+    void ClearSpawnPoints()
+    {
+        spawnPoints.Clear();
+    }
+
+    public void AddSpawnPoint(PlayerSpawnPoint sp)
+    {
+        spawnPoints.Add(sp);
+    }
+
+    //Teleports/'Spawns' the player at the specified spawn point in the current scene
+    public void SpawnPlayerAtSpawnPoint(Protagonist player, string spawnPointLabel)
+    {
+        foreach(PlayerSpawnPoint sp in spawnPoints)
+        {
+            if(sp.label == spawnPointLabel)
+            {
+                Spawn(player, sp.transform.position, sp.transform.rotation);
+                return;
+            }
+        }
+    }
+
+    void Spawn(Protagonist player, Vector3 pos, Quaternion rot)
+    {
+        player.GetComponent<CharacterController>().enabled = false; //Disable the controller due to it interfering with changing the position directly
+        player.transform.position = pos;
+        player.GetComponent<CharacterController>().enabled = true;
```

Comment:
Please try to avoid calling `GetComponent` multiple times, it would be better to cache the `CharacterController` object inside the `Protagonist` class itself on Awake. Using `GetComponent` is quite inefficient in general and should be avoided outside of `Awake` or `Start` if at all possible, calling it twice in a row would already warrant storing a local variable for it at least as there's no use to do the same lookup twice